src/components/BlogPost.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import { ListGroupItem, Collapse } from 'reactstrap';
+import BlogContextMenu from './BlogContextMenu';
+import CommentsSection from './CommentsSection';
+import CommentEntry from './CommentEntry';
+
+class BlogPost extends React.Component {
+  constructor(props) {
+    super(props);    
+    
+    this.onPost = this.onPost.bind(this);
+    this.submitEdit = this.submitEdit.bind(this);
+
+    this.childElement = React.createRef();
+  }
+  
+  error(message) {
+    console.log(message);
+  }
+
+  onPost(comment)
+  {
+    comment.index = this.props.post.comments.length;
+    this.props.post.comments.push(comment);
+    this.childElement.current.expand();
+  }
+
+  saveEdit(comments, position, comment)
+  {      
+    if(position.length == 1)
+    {
+        comments[position[0]] = comment;
+    }
+    else if(position.length > 1)
+    {
+        this.saveEdit(comments[position[0]].comments, position.slice(1), comment);
+    }
+  }
+
+  async submitEdit(comment, position)
+  {
+    //update current state
+    this.saveEdit(this.props.post.comments, position, comment);
+  }
+
+  render() {
+    var post = this.props.post;
+    return (
```

Comment:
Can you make sure that the indentation is consistently two spaces here? Although maybe this comment should just wait until we have prettier set up.

Reply:
I'll wait for prettier - there is a lot of inconsistency between files for whitespace